### ISSUE TITLE
`EResult.LookbehindOverrun` and `EResult.InputOverrun`

### ISSUE THREAD

**Issue by AT0myks** [vlaci/lzallright#20]
I wanted to bump ubi_reader to v0.8.8 for a project, but I'm getting errors for all of my UBIFS test files, which make it impossible to extract files from the image. If I change ubi_reader to use python-lzo instead, no errors. This is with ubi_reader v0.8.8 and lzallright v0.2.2. Here's a [test file](https://github.com/vlaci/lzallright/files/11583168/07_app.zip). Just run `ubireader_extract_files 07_app.bin`. I'm obviously not gonna post the 139 images but my guess is that this is the same error(s) for all of them so hopefully one is enough.

**Comment by vlaci** [comment]
It could be an implementation quirk that `lzokay` doesn't support reading the uncompressed length at the begining of the buffer.

Can you test, if this works for you?

```diff
diff --git a/ubireader/ubifs/misc.py b/ubireader/ubifs/misc.py
index 0368369..3cbfa31 100755
--- a/ubireader/ubifs/misc.py
+++ b/ubireader/ubifs/misc.py
@@ -62,7 +62,7 @@ def decompress(ctype, unc_len, data):
     """
     if ctype == UBIFS_COMPR_LZO:
         try:
-            return lzo.decompress(b''.join((b'\xf0', struct.pack('>I', unc_len), data)))
+            return lzo.decompress(data, output_size_hint=unc_len)
         except Exception as e:
             error(decompress, 'Warn', 'LZO Error: %s' % e)
     elif ctype == UBIFS_COMPR_ZLIB:
```

**Comment by AT0myks** [comment]
That worked, with and without the size hint. I also checked that all calls give the same result as `lzo.decompress(data, False, unc_len)` from python-lzo. I should have thought about trying this first before opening the issue.

**Comment by eabase** [comment]
I'm looking at the same issue. Is there a PR for resolving this?

**Comment by vlaci** [comment]
> I'm looking at the same issue. Is there a PR for resolving this?

Opened https://github.com/onekey-sec/ubi_reader/pull/86

**Comment by vlaci** [comment]
Fixed in [ubi_reader 0.8.9](https://pypi.org/project/ubi-reader/0.8.9/)
